openmdao/core/system.py

```diff
@@ -2967,10 +2970,12 @@ def get_design_vars(self, recurse=True, get_sizes=True, use_prom_ivc=True):
                     out.update(dvs)
 
             if self.comm.size > 1 and self._subsystems_allprocs:
-                allouts = self.comm.allgather(out)
-                out = OrderedDict()
-                for all_out in allouts:
-                    out.update(all_out)
+                for name in out:
```

Comment:
I think with the change above to the 'size' for distributed vars, the old loop here should work fine.  The new loop has a number of problems.  Try reverting to the old one now that the distrib size will be fixed and see if it works.